edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -1056,11 +1060,25 @@ private List getTotalPointsEarnedInternal(final String studentId, final Gradeboo
 								if (!excused) {
 									assignmentsTaken.add(go.getId());
 									literalTotalPointsEarned = pointsEarned.add(literalTotalPointsEarned, GradebookService.MATH_CONTEXT);
+
+									// If category is equal weight, manipulate points to be the average
+									if (cate.isEqualWeightAssignments()) {
+										pointsEarned = pointsEarned.divide(pointsPossible, GradebookService.MATH_CONTEXT);
+									}
+
 									if (cateScoreMap.get(cate.getId()) != null) {
 										cateScoreMap.put(cate.getId(), ((BigDecimal)cateScoreMap.get(cate.getId())).add(pointsEarned, GradebookService.MATH_CONTEXT));
 									} else {
 										cateScoreMap.put(cate.getId(), pointsEarned);
 									}
+
+									// Running average for an equal-weight category
+/*									final BigDecimal singleItemAverage = pointsEarned.divide(pointsPossible, GradebookService.MATH_CONTEXT);
+									if (cateMeanScoreMap.get(cate.getId()) != null) {
+										cateMeanScoreMap.put(cate.getId(), cateMeanScoreMap.get(cate.getId()).add(singleItemAverage, GradebookService.MATH_CONTEXT));
+									} else {
+										cateMeanScoreMap.put(cate.getId(), pointsEarned);
+									}*/
```

Comment:
Can we just remove this block that's commented out?

Reply:
yes agree